src/process/terminalsource.rs

```diff
@@ -107,10 +108,16 @@ impl ColorableTerminal {
             #[cfg(all(test, feature = "test"))]
             StreamSelector::TestTtyWriter(w) => TerminalInner::TestWriter(w, choice),
         };
+        let width = process
```

Comment:
This can be simplified by deferring the conversion from `Result` to `Option` to later, I think:

```rust
let width = process
     .var("RUSTUP_TERM_WIDTH")
     .and_then(|s| u16::from_str(&s))
     .ok()
     .filter(|&n| n > 0);
```

Avoiding the turbofish by using the direct trait method also seems nicer to me.

Reply:
@djc That's not the same kind of error, I guess, which makes them not collapsible?